### ISSUE TITLE
🐛 [firebase_auth] error.credential is always null, OAuthProvider.credentialFromError doesn't exist

### ISSUE THREAD

**Issue by dsgriffin** [FirebaseExtended/flutterfire#11164]
## Bug report

When attempting account linking in cases of `email-already-in-use`, I noticed that `error.credential` is now always `null`.

```
  catch (error) {
      if (error is FirebaseAuthException && error.code == 'email-already-in-use') {
        await auth.signInWithCredential(error.credential!); // error.credential is now always null
      }
  ...
```

I can see the `error.credential` prop [has been removed](https://firebase.google.com/docs/reference/js/auth.autherror) from `AuthError`, but it still shown as existing in `FirebaseAuthException` in the `firebase_auth` package.

[Here](https://github.com/firebase/firebase-js-sdk/issues/5057#issuecomment-867129996) we are told by one of the Firebase team to replace usage of `error.credential` with the [OAuthProvider.credentialFromError](https://firebase.google.com/docs/reference/js/auth.oauthprovider.md#oauthprovidercredentialfromerror) method.

However, `credentialFromError` doesn't exist on any of the `OAuthProvider` interfaces in `firebase_auth`. 

### Expected behavior

1) `OAuthProvider.credentialFromError` should exist in `firebase_auth`
2) `FirebaseAuthException` should remove `credential` and match the `AuthError` interface
3) The docs should be updated to show this new correct method of retrieving an error object's credential. 

### Versions

`firebase_auth: ^4.6.1`

**Comment by darshankawar** [comment]
Thanks for the report and analysis.

/cc @Lyokone for thoughts.

**Comment by MirceaX2Mobile** [comment]
This is happening for me too. The credential property is always **null**

**Comment by SteadyAction** [comment]
same here

**Comment by SteadyAction** [comment]
@Lyokone @darshankawar  any update?

**Comment by gazialankus** [comment]
Downgrading firebase_auth to 4.6.1 bypasses this as far as I can see. I do get the e-mail in the exception itself (e.email).

**Comment by MirceaX2Mobile** [comment]
Do we have any updates on this?

**Comment by kevinrobert3** [comment]
Any updates on this? Blocked on this

**Comment by MirceaX2Mobile** [comment]
I'm blocked on this and I can't update the rest of the libraries because of this.

**Comment by MirceaX2Mobile** [comment]
@russellwheatley do you know anything about a fix for this?

**Comment by MirceaX2Mobile** [comment]
@darshankawar do you know anything about a fix for this?

**Comment by MirceaX2Mobile** [comment]
@russellwheatley I encounter this on mobile platform while auth using SMS for example. If the user is already linked to that phone number I'm not getting the credential from the error to log in to the existing account.

```
      switch (e.code) {
        case "provider-already-linked":
          final credential = e.credential;
          if (credential != null) {
            logInToExistingAccount(credential);
          }
          break;
        case "invalid-credential":
          showError(context.strings.invalidCredentials);
          break;
        case "credential-already-in-use":
          final credential = e.credential;
          if (credential != null) {
            logInToExistingAccount(credential);
          }
          break;
    ```
    
    **e.credential** is always **null**  and user will be stuck.
    
    The last working version of firebase auth is this one:
`firebase_auth: 4.6.2`

**Comment by kevinrobert3** [comment]
@russellwheatley Have encountered this on Apple sign-in and Facebook sign-in when linking providers. This for me is for both iOS and Android

**Comment by MirceaX2Mobile** [comment]
@russellwheatley @darshankawar 
I saw that this fix is released. Now you get the credential, but when you try to use it

`await auth.signInWithCredential(credential);`

The sign in It's not working anymore. You get this

![Screenshot 2023-10-20 at 18 31 14](https://github.com/firebase/flutterfire/assets/33491440/e75420e4-ebbf-4020-870a-3d52866e128d)

**Comment by MirceaX2Mobile** [comment]
@russellwheatley @darshankawar 
Can you look into this please? This is blocking my app for a really long time right now. I can't update the pubs like this :(

**Comment by russellwheatley** [comment]
@MirceaX2Mobile - could you create a new issue as this is a different problem and could you also provide a repro. Thanks.

**Comment by MirceaX2Mobile** [comment]
@russellwheatley 
I will do that. Or maybe is the same issue with this one? 

If it is should I open a new one?
[https://github.com/firebase/flutterfire/issues/11703](https://github.com/firebase/flutterfire/issues/11703)

**Comment by russellwheatley** [comment]
@MirceaX2Mobile - if you're experiencing this issue trying to link an anonymous account with an email account, then it is that issue.

**Comment by MirceaX2Mobile** [comment]
@MirceaX2Mobile In my case it's using Phone Authentication, but i think it has the same core logic. I will mention that there.
